core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -516,6 +517,11 @@ public GeyserSession(GeyserImpl geyser, BedrockServerSession bedrockServerSessio
             this.lecternCache = new ObjectOpenHashSet<>();
         }
 
+        // Initialize generic statistics to 0, so that they appear in the form
+        for (CustomStatistic customStatistic : CustomStatistic.values()) {
```

Comment:
Can this be moved to the AwardStatsPacket and set if `statistics` is empty? Just so we don't initialize a map unless this player is using statistics.

Reply:
I moved it to updateStatistics in GeyserSession.